### PR TITLE
Downloaded content changes

### PR DESCRIPTION
**Pull request by deeppandya**

### CODE REVIEW

**Comment by deeppandya** [comment]
Added @stoletheminerals for security/privacy concerns

File: src/main/java/com/brave/playlist/playback_service/VideoPlaybackService.kt [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        initializeSessionAndPlayer()
+        lastSavedPositionTimer()
+    }
+
+    private fun initializeSessionAndPlayer() {
+        val dataSourceFactory = DataSource.Factory { FileDataSource.Factory().createDataSource() }
```
Comment (deeppandya) [inline src/main/java/com/brave/playlist/playback_service/VideoPlaybackService.kt:111]: Explicitly mentioned `FileDataSource` as DataSource which only works for File Uri

**Review by samartnik — APPROVED** [review]
lgtm

**Comment by SergeyZhukovsky** [comment]
@deeppandya does that PR have anything new or what the actual changes? As from what I see it's basically some kind of a rebase of old commits. Do we need to review everything or just certain commits?

**Comment by deeppandya** [comment]
> @deeppandya does that PR have anything new or what the actual changes? As from what I see it's basically some kind of a rebase of old commits. Do we need to review everything or just certain commits?

@SergeyZhukovsky I have changed the implementation to work with his content differently. This kotlin module is mainly for the UI and player interaction. I would be better if you can briefly go over the changes as I have changed almost every files with new changes. I am sorry for the trouble but there is no easy way for it.

**Review by SergeyZhukovsky — APPROVED** [review]
++

File: src/main/java/com/brave/playlist/fragment/PlaylistFragment.kt [hunk 2]
```diff
@@ -0,0 +1,596 @@
...
+
+package com.brave.playlist.fragment
+
+/*
+ * Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
```
Comment (tapanmodh) [inline src/main/java/com/brave/playlist/fragment/PlaylistFragment.kt]: Two times copyright

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Gradle files
+.gradle/
+build/
+
+# Local configuration file (sdk path, etc)
+local.properties
+
+# IntelliJ
+*.iml
+.idea/
```
Comment (tapanmodh) [inline .gitignore]: nit : newline at the end

File: schemas/com.brave.playlist.local_database.PlaylistDatabase/1.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        "foreignKeys": []
+      }
+    ],
+    "views": [],
+    "setupQueries": [
+      "CREATE TABLE IF NOT EXISTS room_master_table (id INTEGER PRIMARY KEY,identity_hash TEXT)",
+      "INSERT OR REPLACE INTO room_master_table (id,identity_hash) VALUES(42, '97917132ab7ccac87386de3a7faebfce')"
+    ]
+  }
+}
```
Comment (tapanmodh) [inline schemas/com.brave.playlist.local_database.PlaylistDatabase/1.json:66]: nit : newline at the end

File: src/main/res/drawable/ic_button_bg.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this file,
+  ~ You can obtain one at https://mozilla.org/MPL/2.0/.
+  -->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="48dp" />
+    <solid android:color="#4C54D2" />
+</shape>
```
Comment (tapanmodh) [inline src/main/res/drawable/ic_button_bg.xml]: nit : newline at the end

File: src/main/res/drawable/ic_button_unfilled_bg.xml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  ~ You can obtain one at https://mozilla.org/MPL/2.0/.
+  -->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="60dp" />
+    <stroke
+        android:width="1dp"
+        android:color="#5E6175" />
+</shape>
```
Comment (tapanmodh) [inline src/main/res/drawable/ic_button_unfilled_bg.xml:14]: nit : newline at the end

**Review by tapanmodh — CHANGES_REQUESTED** [review]

File: schemas/com.brave.playlist.local_database.PlaylistDatabase/1.json (hunk 4 above)
Comment (deeppandya) [reply schemas/com.brave.playlist.local_database.PlaylistDatabase/1.json:66]: it's an autogenerated file so we don't need to have newline at the end.

**Review by tapanmodh — APPROVED** [review]
lgtm

**Pull request merged**